railties/lib/rails/generators/generated_attribute.rb

```diff
@@ -88,26 +91,25 @@ def reference?(type)
         private
           # parse possible attribute options like :limit for string/text/binary/integer, :precision/:scale for decimals or :polymorphic for references/belongs_to
           # when declaring options curly brackets should be used
-          def parse_type_and_options(type_with_possible_options)
-            return type_with_possible_options, {} unless type_with_possible_options&.match?(TYPE_WITH_OPTIONS_RE)
-
-            case type_with_possible_options
-            when /(string|text|binary|integer)\{(\d+)\}/
-              return $1, limit: $2.to_i
-            when /decimal\{(\d+)[,.-](\d+)\}/
-              return :decimal, precision: $1.to_i, scale: $2.to_i
+          def parse_type_and_options(type, options)
+            return type, {} if options.blank?
+
+            if type.presence_in(%w(string text binary integer)) && options.match?(/^\d+$/)
+              return type.to_sym, limit: options.to_i
+            elsif type.presence_in(%w(decimal numeric)) && options.match?(/^(\d+)[,.-](\d+)$/)
+              precision, scale = options.split(/[,.-]/)
+              return :decimal, precision: precision.to_i, scale: scale.to_i
             end
 
-            type, provided_options = type_with_possible_options.match(TYPE_WITH_OPTIONS_RE).captures
-            attr_options = parse_attr_options(provided_options)
+            attr_options = parse_attr_options(options)
 
             return type, Hash[attr_options].deep_symbolize_keys
           end
 
           def parse_attr_options(options_str)
             # this allows values like for `default` that are strings with commas
             # e.g. `text{default='hello, world!'}`
```

Comment:
This was still referencing the old syntax:

```suggestion
            # e.g. `text{default:'hello, world!'}`
```